File: ruby/testcode/ssti_025.rb
require_relative 'shared'
require 'erb'

class UserTemplate
  def compile(source)
    ERB.new(source)
  end

  def render(source, context = binding)
    compile(source).result(context)
  end
end

# vuln-code-snippet start ruby_ssti_curly_user
def handler(req)
  engine = UserTemplate.new
  output = engine.render(req.param('template'))  # vuln-code-snippet vuln-line ruby_ssti_curly_user
  BenchmarkResponse.html(output)
end
# vuln-code-snippet end ruby_ssti_curly_user

File: php/testcode/ssti_025.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ssti_twig_custom_filter
function ssti025(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('fn');
    $twig = new Twig\Environment(new Twig\Loader\FilesystemLoader('/views'));
    $filter = new Twig\TwigFilter('custom', $input);
    $twig->addFilter($filter); // vuln-code-snippet vuln-line php_ssti_twig_custom_filter
    $html = $twig->render('base.twig', []);
    return BenchmarkResponse::ok($html);
}
// vuln-code-snippet end php_ssti_twig_custom_filter
